Modules/_winapi.c

```diff
@@ -1920,36 +1909,36 @@ static PyMethodDef winapi_functions[] = {
     {NULL, NULL}
 };
 
-static struct PyModuleDef winapi_module = {
-    PyModuleDef_HEAD_INIT,
-    "_winapi",
-    NULL,
-    -1,
-    winapi_functions,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
+static int 
+SetWinAPIConstant(PyObject* d, const char* fmt, const char* constantString, int constant) {
+    PyObject *value = Py_BuildValue(fmt, constant);
+    if (PyDict_SetItemString(d, constantString, value) < 0) {
+        Py_DECREF(value);
+        return -1;
+    }
+
+    Py_DECREF(value);
+    return 0;
+}
 
 #define WINAPI_CONSTANT(fmt, con) \
-    PyDict_SetItemString(d, #con, Py_BuildValue(fmt, con))
+    if (SetWinAPIConstant(d, fmt, #con, con) < 0) return -1
 
-PyMODINIT_FUNC
-PyInit__winapi(void)
+static int winapi_exec(PyObject *m)
 {
-    PyObject *d;
-    PyObject *m;
+    WinApiState *st = winapi_get_state(m);
 
-    if (PyType_Ready(&OverlappedType) < 0)
-        return NULL;
+    st->overlapped_type = (PyTypeObject *)PyType_FromModuleAndSpec(m, &winapi_overlapped_type_spec, NULL);
+    if (st->overlapped_type == NULL) {
+        return -1;
+    }
 
-    m = PyModule_Create(&winapi_module);
-    if (m == NULL)
-        return NULL;
-    d = PyModule_GetDict(m);
+    if (PyModule_AddObject(m, "Overlapped", (PyObject *)st->overlapped_type) < 0) {
```

Comment:
Use PyModule_AddType() as victor comment in here?

Reply:
PS: if you use PyModule_AddObject() in here, you need add the refcount of st->overlapped_type fistly, because PyModule_AddObject() will decrease the refcount  when it successed.

Reply:
@shihai1991 I'll fix the refcount.  Where would I use PyModule_AddType?

Reply:
@shihai1991 I see now, its just a shortcut to using PyModule_AddObject when the object is a type.  Good to know